looprpc/client.proto

```diff
@@ -666,3 +697,124 @@ message LsatToken {
     */
     string storage_name = 8;
 }
+
+message GetLiquidityConfigRequest{}
+
+message LiquidityConfig{
+    /*
+    Set to true if we should include private channels in our balance
+    calculations.
+    */
+    bool include_private = 1;
+
+    /*
+    The target to apply liquidity rules to.
+    */
+    LiquidityTarget target = 2;
+
+    /*
+    A rule to be applied to the target. This value may not be set if setting
+    target to none, and must be supplied otherwise.
+    */
+    LiquidityRule rule = 3;
+}
+
+enum LiquidityTarget{
+    NONE = 0;
+    NODE = 1;
+    PEER = 2;
+    CHANNEL =3;
+}
+
+enum LiquidityRuleType{
+    UNKNOWN = 0;
+    THRESHOLD = 1;
+}
+
+message LiquidityRule{
+    /*
+    Type indicates the type of rule that this message rule represents. Setting
+    this value will determine which fields are used in the message. The comments
+    on each field in this message will be prefixed with the LiquidityRuleType
+    they belong to.
+    */
+    LiquidityRuleType type = 1;
+
+    /*
+    THRESHOLD: Minimum inbound is the minimum inbound percentage beneath which
+    loop outs to acquire inbound liquidity will be recommended when using the
+    threshold rule.
+    */
+    uint32 minimum_inbound = 2;
+
+    /*
+    THRESHOLD: Minimum outbound is the minimum outbound percentage beneath which
+    loop ins to acquire outbound liquidity will be recommended when using the
+    threshold rule.
+    */
+    uint32 minimum_outbound = 3;
+}
+
+message SetLiquidityConfigRequest{
+    /*
+    Config is the desired new config for the liquidity management subsystem.
+    Note that all of the fields in this config will be used to update your
+    current config,so fields that you do not want to change should be set to
+    their current values.
+    */
+    LiquidityConfig config = 1;
+}
+
+message SuggestSwapsRequest{
+}
+
+message SuggestSwapsResponse{
+    /*
+    Rule is the rule that was applied to get these swap recommendations.
+    */
+    LiquidityRule rule = 1;
+
+    /*
+    The target that the rule was applied to.
+    */
+    LiquidityTarget target = 2;
+
+    /*
+    A map of target ID to suggested swaps.
+    */
+    map<string, SwapSuggestion> swaps = 3;
```

Comment:
The reasoning here was that we may want to expand to allow people to set specific rules for a channel/peer. Which would look something like `loop setrule 029f9d70dfdae96333bae711eb23dce5e13188bcbc6463e73eac19a513361afcd6 {rule values}`. I thought it would be nice to display per-target, especially for custom rules, which is why it's exposed, wdyt?

Reply:
Yes it may develop in that direction. But may comment was just about the data type of a multi-formatted string. Isn't there a stronger typed alternative, like multiple maps with different key types?